test-runner/src/main/scala/com/typesafe/spark/test/mesos/RolesSpec.scala

```diff
@@ -1,61 +1,71 @@
 package com.typesafe.spark.test.mesos
 
+import org.apache.spark.SparkContext
 import org.scalatest.Assertions._
 
 import com.typesafe.spark.test.mesos.mesosstate.MesosCluster
 import com.typesafe.spark.test.mesos.framework.runners.RoleConfigInfo
 
 
-trait RolesSpec { self: MesosIntTestHelper =>
+trait RolesSpec extends RoleSpecHelper {
+  self: MesosIntTestHelper =>
 
   def mesosConsoleUrl: String
 
   def cfg: RoleConfigInfo
 
-  runSparkTest("simple count in fine-grain mode with role",
-    "spark.mesos.coarse" -> "false", "spark.mesos.role" -> cfg.role, "spark.cores.max" -> cfg.roleCpus) { sc =>
-    val rdd = sc.makeRDD(1 to 5)
-    val res = rdd.sum()
+  runSparkTest("simple count in fine-grained mode with role",
+    "spark.mesos.coarse" -> "false", "spark.mesos.role" -> cfg.role) { sc =>
+    testRole(sc)
+  }
 
-    assert(15 == res)
+  runSparkTest("simple count in coarse-grained mode with role",
+    "spark.mesos.coarse" -> "true", "spark.mesos.role" -> cfg.role) { sc =>
+    testRole(sc)
+  }
+}
 
-    val m = MesosCluster.loadStates(mesosConsoleUrl)
-    assert(m.sparkFramework.isDefined, "The driver should be running")
+trait RoleSpecHelper {
+  self: MesosIntTestHelper with RolesSpec =>
 
-    if (cfg.role != "*") {
-      // TODO: add message
-      assert(m.slaves.flatMap { x => x.roleResources.map { y => y.roleName } }.contains(cfg.role))
+  def testRole(sc: SparkContext) : Unit = {
 
-      // TODO: add message
-      assert(m.sparkFramework.get.resources.cpu == cfg.roleCpus.toInt)
+    {
```

Comment:
i just use  val m = MesosCluster.loadStates(mesosConsoleUrl) so since i use the same name for variable in a block later on ididnt want someone to be confused with scope resolution anyway i will change it